### PR TITLE
Implementing Continuous Integration using GitHub Actions

### PR DESCRIPTION
**Pull request by Besttd**
After some really profound git trouble, I've included the first try of continuous integration using GitHub Actions.

I've configured it to do a minor default check with no customization following [this tutorial](https://orchid00.github.io/actions_sandbox/packageci.html#actions-for-continuous-integration-checks).

Briefly, the action triggers an install on macOS and calls `rcmdcheck::rcmdcheck(args = "--no-manual", error_on = "error")` on every push and pull request to main/master.

Unfortunately, I cannot test the functionality on "my" branch because we need to push/pull request to trigger the action. As you can see below, this pull request has already triggered the action.

I'll configure actions for our needs later.

### CODE REVIEW

**Review by p-mq — COMMENTED** [review]
@Besttd this looks great so far! A few questions:
- what do you want to change/test before requesting final approval for this pull request?
- What makes you thin the full test is "overkill"? Does it take too long to run? Does it consume too many github actions?
- Since both of us work on Windows, I'd like to at least have a Linux environment in the mix. Maybe we can be clever about doing two checks covering two OS and two versions of R?

**Comment by Besttd** [comment]
I've experimented with the run time of different setups. It was indeed surprised that the basic workflow took >5 minutes.

Generally, the advice is that one OS is enough unless you have OS-specific code or you are a really major package.
For R versions it boils down to what versions you want to support. I've defaulted to the current R release for now.

In both cases, the motivation to limiting yourself to the minimum is to not use up your computing time. Also, you are very likely not going to encounter errors because of different OSs that wouldn't affect your dependencies/base first. 

After testing around I've no defaulted to:
- **R-CMD-check**: Triggered on **P/PR** to `main/master` running `rcmdcheck(--as-cran)` on **Ubuntu 18.04** with the **current R version** + **R devel** -> this should mirror CRANS behavior
- **tests_only**: Triggered on **P/PR** to `any other branch` runing all **tests** on the latest **windows** machine with the **current R version**

If we get into a more stable development phase we can think about adding more OS.

**PS: Please squash merge because I was experimenting with settings. Thanks!**

**Review by p-mq — APPROVED** [review]
Nice job!

**Pull request merged**
